src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1170,4 +1172,116 @@ private function addRobotsIndexSection(ArrayNodeDefinition $rootNode)
             ->end()
         ;
     }
+
+    private function addHttpClientSection(ArrayNodeDefinition $rootNode)
+    {
+        $subNode = $rootNode
+            ->children()
+                ->arrayNode('http_client')
+                    ->info('HTTP Client configuration')
+                    ->canBeEnabled()
+                    ->fixXmlConfig('client')
+                    ->children();
+        $subNode = $this->addHttpClientOptionsSection($subNode);
+        $subNode = $subNode
+                        ->arrayNode('clients')
+                            ->useAttributeAsKey('name')
+                            ->normalizeKeys(false)
+                            ->arrayPrototype()
+                                ->children();
+        $subNode = $this->addHttpClientOptionsSection($subNode);
+        $subNode = $subNode
+                                ->end()
+                            ->end()
+                        ->end()
+                    ->end()
+                ->end()
+            ->end()
+        ;
+    }
+
+    private function addHttpClientOptionsSection(NodeBuilder $rootNode): NodeBuilder
+    {
+        return $rootNode
+            ->integerNode('max_host_connections')
+                ->info('The maximum number of connections to a single host.')
+                ->defaultValue(6)
+            ->end()
+            ->arrayNode('default_options')
+                ->children()
+                    ->scalarNode('auth')
+                        ->info('An HTTP Basic authentication "username:password".')
+                    ->end()
+                    ->arrayNode('query')
+                        ->info('Associative array of query string values merged with URL parameters.')
+                        ->normalizeKeys(false)
+                        ->scalarPrototype()->end()
+                    ->end()
+                    ->arrayNode('headers')
```

Comment:
I think we should allow `header` elements in the XML format (thus need `fixXmlConfig('header')`)

Reply:
on it